lib/bb/middleware/response/bb_parser.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+module BB
+  module Middleware
+    module Response
+      # class responsible for customizing parsing
+      class BBParser < Faraday::Response::Middleware
+        def on_complete(env)
+          return unless env.response_headers['content-type'] =~ /\bjson/
+          # If POST for prescriptions is successful message body is irrelevant
+          # if it was not successul an exception would have already been raised
+          return if env.method == :post
+          env[:body] = parse(env.body) unless env.body.blank?
+        end
+
+        def parse(body = nil)
+          @parsed_json = body
+          @meta_attributes = split_meta_fields!
+          @errors = @parsed_json.delete(:errors) || {}
+
+          data =  parsed_extract_status_list || parsed_health_record_types
+          @parsed_json = {
+            data: data,
+            errors: @errors,
+            metadata: @meta_attributes
+          }
+          @parsed_json
+        end
+
+        def split_meta_fields!
+          updated_at = @parsed_json.delete(:last_updated_time) ||
+                       @parsed_json.delete(:last_updatedtime)
+
+          {
+            updated_at: updated_at,
+            failed_station_list: @parsed_json.delete(:failed_station_list)
+          }
```

Comment:
My understanding from MHV was that failed_station_list could happen anytime one of their downstream dependencies kicks back an error. In this case I suppose it could be MVI or EVSS's evault, or whatever partners are involved in the PHR refresh process.

I think I'll leave this one in for now for consistency with other prescriptions parser, and we'll just have to wait and see when we encounter this as a negative test case. It's still unclear how those interactions might play out.